### PR TITLE
36 mises plasticity with isotropic hardening 1

### PR DESCRIPTION
**Pull request by saifr68**
Von-mises plasticity with non-linear isotropic hardening implemented with incremental strains formulation. Uni-axial monotonic and cyclic loading tests are written to validate the model against analytical solution.

### CODE REVIEW

File: examples/mises_plasticity/test_plasticity.py [hunk 1]
```diff
@@ -0,0 +1,224 @@
...
+
+    # if material behaves linearly under the elastic range with correct slope
+    indices = load + tolerance < matparam["p_y0"]
+    v = (3 * matparam["p_ka"] - 2 * matparam["p_mu"]) / (2 * (3 * matparam["p_ka"] + matparam["p_mu"]))
+    trace = displacement[indices][1] - 2 * v * displacement[indices][1]
+    dev = displacement[indices][1] - trace / 3
+    slope = (matparam["p_ka"] * trace + 2 * matparam["p_mu"] * dev) / displacement[indices][1]
+    assert np.all(abs(np.ediff1d(load[indices]) / np.ediff1d(displacement[indices]) - slope) < 1e-7)
+
+    # ax = plt.subplots()[1]
```
Comment (srosenbu) [inline examples/mises_plasticity/test_plasticity.py:100]: These comments should be removed as well

File: src/fenics_constitutive/solver.py [hunk 2]
```diff
@@ -94,6 +94,8 @@ def __init__(
         form_compiler_options: dict | None = None,
         jit_options: dict | None = None,
     ):
+        print("help, i am in init")
```
Comment (srosenbu) [inline src/fenics_constitutive/solver.py:97]: These prints should be removed

File: src/fenics_constitutive/solver.py [hunk 3]
```diff
@@ -293,14 +295,21 @@ def form(self, x: PETSc.Vec) -> None:
 
             with df.common.Timer("stress_evaluation"):
                 self.stress_1.x.array[:] = self.stress_0.x.array
-                if law.history_dim is not None:
+                history_input = None
+                if isinstance(law.history_dim, int):
```
Comment (srosenbu) [inline src/fenics_constitutive/solver.py:299]: This option was removed in the last PR. I am not sure why it wants to overwrite the stuff on main. Could you merge main into your branch first?

File: examples/mises_plasticity/Mises_plasticity_without_hardening.py [hunk 4]
```diff
@@ -0,0 +1,252 @@
+#Von Mises plasticity with non-linear isotropic hardening
```
Comment (srosenbu) [inline examples/mises_plasticity/Mises_plasticity_without_hardening.py:1]: Is this file even tested? Do we still need it?

**Review by srosenbu — CHANGES_REQUESTED** [review]
The plasticity model and the tests look really nice. I especially like that you also test the cyclic loading;) 
The files might need a little cleanup (removing comments, formatting with Ruff or Black, etc.).

I am not so sure if we need the other 3 files. Isn't the plasticity without hardening a special case for the hardening plasticity? Also if we need it, it should be tested as well.

File: examples/mises_plasticity/test_plasticity.py [hunk 5]
```diff
@@ -0,0 +1,224 @@
+from __future__ import annotations
+
+import dolfinx as df
+import numpy as np
+from dolfinx.nls.petsc import NewtonSolver
+from mises_plasticity_isotropic_hardening import VonMises3D
+from mpi4py import MPI
+# import matplotlib.pyplot as plt
```
[the remark this answers is not in the document]
Comment (saifr68) [reply examples/mises_plasticity/test_plasticity.py]: Done

File: examples/mises_plasticity/test_plasticity.py (hunk 1 above)
Comment (saifr68) [reply examples/mises_plasticity/test_plasticity.py]: Done

**Review by srosenbu — APPROVED** [review]
Nice:)
Please merge!

**Pull request merged**